cli/src/cmd/forge/script/executor.rs

```diff
@@ -134,22 +153,11 @@ impl ScriptArgs {
                         )
                         .expect("Internal EVM error");
 
-                    // Identify all contracts created during the call.
-                    if result.traces.is_empty() {
-                        eyre::bail!(
-                            "Forge script requires tracing enabled to collect created contracts."
-                        )
-                    }
-
-                    if !result.success || script_config.evm_opts.verbosity > 3 {
-                        for (_kind, trace) in &mut result.traces {
-                            decoder.decode(trace).await;
-                            println!("{}", trace);
-                        }
-                    }
-
-                    if !result.success {
-                        eyre::bail!("Simulated execution failed");
+                    if !result.success || result.traces.is_empty() {
+                        return Ok::<
+                            (Option<TransactionWithMetadata>, Vec<(TraceKind, CallTraceArena)>),
+                            eyre::ErrReport,
+                        >((None, result.traces))
```

Comment:
I'd rather move this hint to the `futs` directly:

`let futs: Vec<....> = ...`